### ISSUE TITLE
Incorrect word endings with `generate`

### ISSUE THREAD

**Issue by csillag** [hunspell/hunspell#457]
Hello,

I am new to Hunspell. I am trying to generate some Hungarian text, and I am using Hunspell to generate the required word endings for this.

I have this very simple utility, which basically exposes Hunspell's `generate` function:

``` c++
#include <string.h>
#include <stdlib.h>
#include <stdio.h>

#include <hunspell.hxx>

int main(int argc, char** argv)
{
  Hunspell * pH;
  if (argc > 2) {
    pH = new Hunspell("/usr/share/hunspell/hu_HU.aff", "/usr/share/hunspell/\
hu_HU.dic", (const char *) NULL);
  } else {
    pH = new Hunspell(argv[3], argv[4], (const char *) NULL);
  }

  char ** gen;
  int l = pH->generate(&gen, argv[1], argv[2]);
  if (l) {
    printf("%s\n", gen[0]);
  } else {
    printf("%s\n", argv[1]);
  }
  return 0;
}
```

I compile this with Hunspell 1.4.1, and I use the dictionary files from 5.2.4.

This is what the tool returns:
 * `./ragozo zöld kékek`  =>  `zöldek` (so far, so good)
 * `./ragozo sárga kékek` => `sárgák` (wonderful)
 * `./ragozo kicsi kékek` => `kicsik` (no surprise here)

So, basically it works. But then:

 * `./ragozo nagyfejű kékek` => `nagyfejűk`

This is not correct; the expected result is "nagyfejű**e**k".

Is this a bug, or am I doing something wrong, or have I reached the limits of the current iteration of word ending generation of Hunspell? Thank you for explaining.

**Comment by laszlonemeth** [comment]
Hello,

This is a known problem of the 

(1) correct weightening of the same category suffixes at morphological generation
(2) combined with a limitation here to avoid substandard forms.

Affix file of the Hungarian dictionary defines a GENERATE flag to handle this, but only the similar SUBSTANDARD flag is used by the recent Hunspell. Unfortunately, these plural forms can be substandard or not depending from the stem or from the previous suffixes, and they need more complex morphological and dictionary descriptions, than the recent ones.
I can imagine a better solution to remove the artificial limit in the morphological generation, and checking the frequency of the plurals in the frequency word list of the Hungarian Webcorpus. Next step could be to extend the dictionary based on the corpus.

Thanks for your bug report.

Best regards,
László
